sos/plugins/postgresql.py

```diff
@@ -47,34 +48,34 @@ class PostgreSQL(Plugin):
         ('dbport', 'database server port number', '', '5432')
     ]
 
-    def pg_dump(self, pg_dump_command="pg_dump", filename="sos_pgdump.tar"):
+    def pg_dump(self, scl=None):
         if self.get_option("dbname"):
             if self.get_option("password") or "PGPASSWORD" in os.environ:
                 self.tmp_dir = tempfile.mkdtemp()
```

Comment:
Why are we using `tempfile` directly here and not simply directing the output into the desired path in the report archive?

It makes no sense to generate a file by running a command in `setup()` and then collect it: it should just be written to the report in the first place.

This is an anti-pattern we've been eliminating for years, first in Gluster and the various RHEV plugins, and now here... :-/

I realise this wasn't introduced in these patches but it needs to go, and soon (along with the `postproc()` cleanup).